src/utils/screenshots.js

```diff
@@ -0,0 +1,65 @@
+const getCustomScreenshots = (ticket, defaultUserAgent) => {
+  // pull the fields out of the screenshot object into something a bit easier to work with:
+  const ticketScreenshots = ticket.get('ScreenShots').map(ss => ({
+    ticketId: ss.get('TicketId'),
+    width: ss.get('width'),
+    height: ss.get('height'),
+    filename: ss.get('filename'),
+    userAgent: ss.get('userAgent'),
+  }));
+
+  return ticketScreenshots;
+};
+
+const processScreenshot = async (page, ss, defaultUserAgent) => {
+  console.log(`processing screenshot for ticket: ${ss.ticketId}`);
+
+  // set viewport
+  await page.setViewport({
+    width: ss.width,
+    height: ss.height,
+    deviceScaleFactor: 1,
+  });
+
+  // set user agent
+  const userAgent = ss.userAgent || defaultUserAgent;
+  page.setUserAgent(userAgent);
+
+  // capture screenshot
+  const data = await page.screenshot();
+
+  // data object is a buffer, we'll convert it later:
+  artifact = {
+    screenshot: ss,
+    data: data,
+  };
+
+  return artifact;
+};
+
+const processFullScreenshot = async (page, defaultUserAgent, ticketId) => {
+  console.log('\tProcessing fullpage screenshot');
+
+  // initialize fullpage screenshot object
+  const ss = {
+    ticketId: ticketId,
+    filename: 'screenshotFull.png',
+    userAgent: defaultUserAgent,
+  };
+
+  // capture screenshot
+  const data = await page.screenshot({ fullPage: true });
+  // create artifact object to return
+  const artifact = {
+    screenshot: ss,
+    data: data,
```

Comment:
The only functionality I changed was removing the call to set the viewport. Originally, the full page screenshot was setting the viewport to a predefined height and width, which was messing up with capturing the full page. All that is needed is the `fullPage: true` setting

Reply:
Because the custom screenshots are being processed before the full page ones, and set the viewport for each one, when the full page screenshot is taken it will have the viewport from whatever the last custom screenshot taken was. If it was messing up the full page screenshot before, what did they look like? There's an `await` call before `setViewport`, so it should be finished before the screenshot is taken.

Reply:
@williamtdr ohhhh that's a good point. The custom screenshots are not being processed correctly, so I wasn't able to see this happen, but that seems like a likely outcome. I can move the fullpage screenshot to be first. Previously for taking a screenshot of google.com, the fullpage screenshot (with a specified viewport) was 40 KB, but after taking the viewport setting out it was 32 KB.